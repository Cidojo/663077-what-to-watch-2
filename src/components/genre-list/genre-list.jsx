import * as React from 'react';
import * as PropTypes from 'prop-types';
import {GenreItem} from './../genre-item/genre-item.jsx';

const GenreList = (props) => {
  const {
    currentGenre,
    genres,
    maxGenresToDisplay,
    onGenreTabClick
  } = props;

  return (
    <ul className="catalog__genres-list">
      <GenreItem
        currentGenre={currentGenre}
        genre={`All genres`}
        onGenreTabClick={onGenreTabClick}
      />
      {
        genres.slice(maxGenresToDisplay).map((genre) => {
          return (
            <GenreItem
              key={genre}
              currentGenre={currentGenre}
              genre={genre}
              onGenreTabClick={onGenreTabClick}
            />
          );
        })
      }
    </ul>
  );
};

GenreList.propTypes = {
  currentGenre: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
  maxGenresToDisplay: PropTypes.number,
  onGenreTabClick: PropTypes.func.isRequired,
};

export {GenreList};
