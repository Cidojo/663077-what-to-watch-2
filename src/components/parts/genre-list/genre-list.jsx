import * as React from 'react';
import * as PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {GenreItem} from './../genre-item/genre-item.jsx';

import {GenreActionCreator} from './../../../reducers/genre-reducer/genre-reducer.js';
import {DisplayCountActionCreator} from './../../../reducers/display-count-reducer/display-count-reducer.js';
import Selectors from './../../../selectors/selectors.js';
import {SHOW_MORE_STEP} from './../../../constants/constants.js';

const GenreList = (props) => {
  const {
    activeGenre,
    genres,
    onGenreChange
  } = props;

  return (
    <ul className="catalog__genres-list">
      {
        genres.map((genre) => {
          return (
            <GenreItem
              key={genre}
              isActive={activeGenre === genre}
              genre={genre}
              onGenreTabClick={onGenreChange}
            />
          );
        })
      }
    </ul>
  );
};

GenreList.propTypes = {
  activeGenre: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
  onGenreChange: PropTypes.func.isRequired,
};

GenreList.defaultProps = {
  activeGenre: ``,
  genres: [],
  onGenreChange: () => {}
};

const mapDispatchToProps = (dispatch) => ({
  onGenreChange: (genre) => {
    dispatch(GenreActionCreator.assignGenre(genre));
    dispatch(DisplayCountActionCreator.setCount(SHOW_MORE_STEP));
  }
});

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    activeGenre: Selectors.getGenre(state),
    genres: Selectors.getGenresList(state)
  });
};

export {GenreList};
export default connect(mapStateToProps, mapDispatchToProps)(GenreList);
