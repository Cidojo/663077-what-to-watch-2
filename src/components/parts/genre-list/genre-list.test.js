import * as React from 'react';
import {GenreList} from './genre-list.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  activeGenre: ``,
  genres: [],
  onGenreChange: () => {}
};

it(`should render GenreList component without errors`, () => {
  const tree = renderer.create(<GenreList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
