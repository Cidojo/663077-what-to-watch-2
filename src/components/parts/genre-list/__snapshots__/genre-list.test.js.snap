// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GenreList component without errors 1`] = `
<ul
  className="catalog__genres-list"
/>
`;
